File: apps/jetbrains-plugin/src/main/kotlin/com/windsurf/controller/services/detection/detectors/AiderDetector.kt
package com.windsurf.controller.services.detection.detectors

import com.windsurf.controller.services.detection.AgentDetector
import com.windsurf.controller.services.detection.DetectionContext
import com.windsurf.controller.services.detection.DetectionResult
import com.windsurf.controller.services.detection.checks.expandHome
import com.windsurf.controller.services.detection.checks.fileExists
import com.windsurf.controller.services.detection.checks.whichBinary

/**
 * Aider — terminal agent owned by codeam-cli. No JetBrains plugin
 * exists; detection is the `aider` binary on PATH or the single-file
 * config `~/.aider.conf.yml`.
 */
class AiderDetector : AgentDetector {
    override val id = "aider"
    override val name = "Aider"
    override val icon = "aider"

    private val binaryName = "aider"
    // Aider config-file locations, in priority order. All three are
    // documented in Aider's README and work cross-platform — the
    // `~/.config/aider/` variant is the XDG-spec path Linux users
    // tend to prefer, the `~/.aider.conf.yml` form is the historical
    // default that also resolves correctly on Windows
    // (`C:\Users\<user>\.aider.conf.yml`).
    private val configFiles = listOf(
        "~/.aider.conf.yml",
        "~/.config/aider/aider.conf.yml",
    )

    override suspend fun detect(ctx: DetectionContext): List<DetectionResult> {
        val configPresent = configFiles.any { fileExists(expandHome(it)) }
        val installed = whichBinary(binaryName) != null || configPresent
        if (!installed) return emptyList()
        ctx.logger.info("[AiderDetector] detected")
        return listOf(
            DetectionResult(
                id = id,
                name = name,
                icon = icon,
                pluginId = "aider.aider",
                toolWindowId = "__terminal__:$id",
                isTerminalAgent = true,
            ),
        )
    }
}

File: apps/vsc-plugin/src/services/agent-detection/detectors/aider.detector.ts
import { whichBinary } from '../checks/binary';
import { fileExists, expandHome } from '../checks/config-dir';
import type { AgentDetector, DetectionContext, DetectionResult } from '../types';

/**
 * Detects Aider (paul-gauthier/aider — interactive coding agent that
 * pairs an LLM with the user's git repo). Two signals, first hit wins:
 *
 *   1. The `aider` binary on PATH (typical `pip install aider-chat` /
 *      pipx install).
 *   2. `~/.aider.conf.yml` (a single YAML file, not a directory) —
 *      Aider's own config home which is also where its API-key
 *      escape hatch is documented.
 *
 * No VS Code extension probe — Aider has no official marketplace
 * extension. Every branch returns `isTerminalAgent: true` so the
 * mobile dispatches runtime commands to codeam-cli's pluginId.
 */
export class AiderDetector implements AgentDetector {
  readonly id = 'aider';
  readonly name = 'Aider';
  readonly icon = 'aider';

  private static readonly BINARY_NAME = 'aider';
  // Aider config-file locations, both documented in Aider's README and
  // cross-platform: `~/.aider.conf.yml` is the historical default
  // (resolves to `C:\Users\<user>\.aider.conf.yml` on Windows),
  // `~/.config/aider/aider.conf.yml` is the XDG variant Linux users
  // tend to prefer.
  private static readonly CONFIG_FILES = ['~/.aider.conf.yml', '~/.config/aider/aider.conf.yml'];

  async detect(ctx: DetectionContext): Promise<DetectionResult | null> {
    const binPath = await whichBinary(AiderDetector.BINARY_NAME);
    if (binPath) {
      ctx.log.appendLine(`[AiderDetector] binary at ${binPath}`);
      return {
        installed: true,
        extensionId: `__binary__:${AiderDetector.BINARY_NAME}`,
        isTerminalAgent: true,
        via: 'binary',
      };
    }

    for (const file of AiderDetector.CONFIG_FILES) {
      if (await fileExists(expandHome(file))) {
        ctx.log.appendLine(`[AiderDetector] config file ${file} present`);
        return {
          installed: true,
          extensionId: `__config__:${AiderDetector.BINARY_NAME}`,
          isTerminalAgent: true,
          via: 'config-dir',
        };
      }
    }

    return null;
  }
}
